### ISSUE TITLE
Latest netlify deploy has new warning message "calling IsSet with unsupported type"

### ISSUE THREAD

**Issue by deining** [google/docsy#901]
With latest commit 66818eb0905c014f342432c5fc918fc4b1c04b57 in place, I now see a warning when serving the user guide.

```
8:27:49 PM: > hugo --cleanDestinationDir --themesDir ../.. "--minify"
8:27:49 PM: Start building sites …
8:27:49 PM: hugo v0.92.2-CDF6A0D6+extended linux/amd64 BuildDate=2022-02-11T14:17:39Z VendorInfo=gohugoio
8:27:50 PM: WARNING: calling IsSet with unsupported type "invalid" (<nil>) will always return false.
```

This warning can be seen also in the [netlify log](https://app.netlify.com/sites/docsydocs/deploys/620ff2f447456d000765ce21) for this commit (line 95).

@raum51: is this warning related to the changes you submitted with #895?

**Comment by raum51** [comment]
I'm afraid, that its related to my changes from #895, because I've build in isset on 2 places. Sorry.
On my test system there was no problems.

I'll take a look.

**Comment by raum51** [comment]
But the issue with isset rises only a warning and in the next line of the log it shows an error:

8:27:53 PM: Error: Error building site: POSTCSS: failed to transform "scss/main.css" (text/css). Check your PostCSS installation; install with "npm install postcss-cli". See https://gohugo.io/hugo-pipes/postcss/: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information

Look more like a problem with SCSS/CSS, I think.

**Comment by deining** [comment]
> But the issue with isset rises only a warning and in the next line of the log it shows an error:

These are 2 different, unrelated issues:

- a warning, which doesn't prevent successful deploy of the user guide. We should get rid of this warning, though.
- a failure, adressed in #900.

For me, these are separate, independent issues, that's why a raised two separate issues (#900 / #901).
Thanks for investigating the warning message adressed with this ticket.

**Comment by chalin** [comment]
The build failure was my bad, and has been fixed. Thanks for reporting those @deining.

**Comment by raum51** [comment]
Thanks @chalin for the info.

So also the warning caused by my PR #895 is solved or do I have to check it?

**Comment by chalin** [comment]
> So also the warning caused by my PR #895 is solved or do I have to check it?

The warning is still there, for example, see the [latest production log](https://app.netlify.com/sites/docsydocs/deploys/62140dcbd44e830008db6061#L119).

**Comment by raum51** [comment]
OK, I'll take a look after official work ;-)

**Comment by raum51** [comment]
I think, I know why you get the warning. But I can't figure out, why I DON'T get the warning on my system. Hugo is building the site without any warning. I'm using hugo v0.92.2+extended darwin/amd64 on a MAC.

Have anyone an explanation for that?

**Comment by chalin** [comment]
I'm using the same version of Hugo under macOS. Which command are you using to build?

**Comment by raum51** [comment]
OK, I've found it. Was my fault: haven't build the userguide but my own site with dev content.

**Comment by raum51** [comment]
Oh, there's a long story I'd completely forgotten.

- I've made my PR #895 because of the issue #888: Hiding taxonomy cloud by setting `taxonomyCloud = []` didn't work.
- Hiding didn't work since https://github.com/google/docsy/commit/9759ccdf4326cdcca6a67a0ced184128660acfd2#diff-e6e949b24d2c6cb1724f689ca2adbe3257e4fe1df503d3182ac9a28d3bc719f7
- In #895 I changed it back to the old version, which makes incorrect use of isset
==> A never ending story!

The problem is, I want to control 3 different states with one parameter `.Site.Params.Taxonomy.taxonomyCloud `
- default: show all defined taxonomies, when taxonomyCloud is not set
- hide the taxonomy cloud by `taxonomyCloud = []`
- show only specific clouds - e.g. `taxonomyCloud = ["tags", "categories"]`
==> Can't find a way to distinguish between not set and empty array. 


**The easiest way I see would be to set at least the parameter group `[params.taxonomy]` in our config.toml:**
```
[params.taxonomy]
# taxonomyCloud = ["tags", "categories"] # set taxonomyCloud = [] to hide taxonomy clouds
# taxonomyCloudTitle = ["Tag Cloud", "Cloud of Categories"] # if used, must have same lang as taxonomyCloud
# taxonomyPageHeader = ["tags", "categories"] # set taxonomyPageHeader = [] to hide taxonomies on the page headers
```

instead of
https://github.com/google/docsy/blob/f03cba057410f5f94afb5eb21059c3874373e0c4/userguide/config.toml#L231-L234


**==> Would this be OK? Or have anybody any other solution?**

**Comment by chalin** [comment]
I would not force projects to set a field.

I believe that use of `isset` is enough help you to distinguish between the cases of `taxonomyCloud` being set or not. You just might need a more complex if condition.

**Comment by raum51** [comment]
> Btw, I _think_ that the problem is because you're testing whether a field of `params.taxonomy` is set before testing whether `params.taxonomy` is itself defined.

I've already thought about that possibility, but then I have to double the code for FALSE:

```go
{{ $context := . }}
{{ if isset .Site.Params "taxonomy" }}
	{{ if isset .Site.Params.Taxonomy "taxonomycloud" }}
		{{ range $index, $taxo := .Site.Params.Taxonomy.taxonomyCloud }}
			{{ if isset $.Site.Params.Taxonomy "taxonomycloudtitle" }}
				{{ $.Scratch.Set "title" (index $.Site.Params.Taxonomy.taxonomyCloudTitle $index) }}
			{{ else }}
				{{ $.Scratch.Set "title" (humanize $taxo) }}
			{{ end }}
			{{ partial "taxonomy_terms_cloud.html" (dict "context" $context "taxo" $taxo "title" ($.Scratch.Get "title")) }}
		{{ end }}
	{{ else }}
		{{ range $taxo, $taxo_map := .Site.Taxonomies }}
			{{ partial "taxonomy_terms_cloud.html" (dict "context" $context "taxo" $taxo "title" (humanize $taxo)) }}
		{{ end }}
	{{ end }}
{{ else }}
	{{ range $taxo, $taxo_map := .Site.Taxonomies }}
		{{ partial "taxonomy_terms_cloud.html" (dict "context" $context "taxo" $taxo "title" (humanize $taxo)) }}
	{{ end }}
{{ end }}
```

But for me it's OK. I'll prepare the PR.

**Comment by chalin** [comment]
Closed by #909
